### ISSUE TITLE
Delay when calling self.performSegue from .onCellSelection

### ISSUE THREAD

**Issue by BertPorter** [xmartlabs/Eureka#1532]
Environment: Eureka 4.1.1, Xcode 9.3 (9E145) and iOS 11.3 

I have had this problem consistently over the last couple months. On all devices on the Simulator plus iPad Pro and iPhone SE.
When I tap a row sometimes it brings up a segued view controller; other times it doesn't. If I tap twice on the row, is always brings up the segued view controller.

I decided to delve into this tonight as the rest of the app is complete.

What I discovered is that is always invokes the segue. But sometimes this occurs immediately and others times there is a delay ranging from a few seconds to even a minute. And if I tap a second time, it is always immediate.

Here are the results of 4 tests from my iPhone SE, each time with a fresh build from XCode.
I put in some print statements with time stamps.

 First Time Running App – 23 seconds delay
.onCellSelection    20 : 15 : 37
prepare for segue    20 : 15 : 37
.viewDidLoad    20 : 16 : 0

Second Time Running App – 4 seconds delay
.onCellSelection    20 : 17 : 48
prepare for segue    20 : 17 : 48
.viewDidLoad    20 : 17 : 52

Third Time Running App - immediate
.onCellSelection    20 : 18 : 31
prepare for segue    20 : 18 : 31
.viewDidLoad    20 : 18 : 31

Fourth Time Running App – 14 seconds delay
.onCellSelection    20 : 19 : 8
prepare for segue    20 : 19 : 8
.viewDidLoad    20 : 19 : 22

Relevant parts of my code:

     DecimalRow.defaultCellSetup = { cell, row in
            cell.height = { 34.0 }
            cell.textLabel?.font = UIFont.boldSystemFont(ofSize: 14)
            cell.textLabel?.minimumScaleFactor = 10.0/14.0
            cell.textLabel?.adjustsFontSizeToFitWidth = true
            row.formatter = setIntegerNumberFormatter()
        }
        
        DecimalRow.defaultCellUpdate = { cell, row in
            cell.textField.font = UIFont.boldSystemFont(ofSize: 14)
            cell.textField.keyboardType = .numberPad
            NKInputView.with(cell.textField, type: .numberPad, returnKeyType: .next)
            row.useFormatterOnDidBeginEditing = true
        }

        <<< DecimalRow(masterViewTags.emptyWt) {row in
                row.title = row.tag
                row.value = WTBLData.sharedInstance.wTBLPlistItems.emptyWt
                row.disabled = true
                } .onCellSelection{cell, row in
                    
                    let date = Date()
                    let calendar = Calendar.current
                    let hour = calendar.component(.hour, from: date)
                    let minutes = calendar.component(.minute, from: date)
                    let seconds = calendar.component(.second, from: date)
                    print(".onCellSelection    \(hour) : \(minutes) : \(seconds)")
                    
                    self.performSegue(withIdentifier: MasterViewSegues.emptyWt, sender: self)
                    
                }.cellUpdate{ cell, row in
                    row.cell.textLabel?.textColor = UIColor.orange
                    row.cell.textField.textColor = UIColor.orange
                    row.cell.height = { 38 }
            }

Note: This specific row is used for displaying data not editing inline. That is why it is disabled.

**Comment by mats-claassen** [comment]
Does any of these threads help you?

https://forums.developer.apple.com/thread/5861
https://stackoverflow.com/questions/39586785/delay-in-performsegue

**Comment by BertPorter** [comment]
Thank you for your quick response.
I looked at both links.
First link:
They are providing a workaround for a bug in iOS 8. Who knows if this bug still exists? If I add the call to didSelectRowAtIndexPath that is recommended would this interfere with Eureka in any way? The code they use no longer works for iOS 11 so it would take me some time to figure out how to implement it.
Second link:
I tried commenting out all the code in ViewDIdLoad and ViewDidAppear on the segued view controller. This didn't make a difference.
Thanks again.
